otelcolbuilder/cmd/collector_windows.go

```diff
@@ -220,20 +224,25 @@ func withWindowsCore(elog *eventlog.Log) func(zapcore.Core) zapcore.Core {
 	}
 }
 
-func newCollectorWithFlags(set otelcol.CollectorSettings, flags *flag.FlagSet) (*otelcol.Collector, error) {
+func updateSettingsUsingFlags(set *otelcol.CollectorSettings, flags *flag.FlagSet) error {
 	if set.ConfigProvider == nil {
+		resolverSet := &set.ConfigProviderSettings.ResolverSettings
 		configFlags := getConfigFlag(flags)
-		if len(configFlags) == 0 {
-			return nil, errors.New("at least one config flag must be provided")
-		}
 
-		var err error
-		set.ConfigProvider, err = otelcol.NewConfigProvider(newDefaultConfigProviderSettings(configFlags))
-		if err != nil {
-			return nil, err
+		if len(configFlags) > 0 {
+			resolverSet.URIs = configFlags
+		}
+		if len(resolverSet.URIs) == 0 {
+			return errors.New("at least one config flag must be provided")
+		}
```

Comment:
I would change the order of validation

```suggestion
		if len(configFlags) == 0 {
			return errors.New("at least one config flag must be provided")
		}
		if len(configFlags) > 0 {
			resolverSet.URIs = configFlags
		}
```

Reply:
@sumo-drosiek - Agreed, however this code was copied directly from [upstream](https://github.com/open-telemetry/opentelemetry-collector/blob/4627d3c18ea8cd82061a70f8db30ef98426e77db/otelcol/command.go#L44) so I left it alone.